zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Note.java

```diff
@@ -365,6 +369,13 @@ public void fireParagraphUpdateEvent(Paragraph p) throws IOException {
    */
   public Paragraph insertNewParagraph(int index, AuthenticationInfo authenticationInfo) {
     Paragraph paragraph = new Paragraph(this, paragraphJobListener);
+    if (null != interpreterSettingManager) {
+      // Set the default parameter configuration for the paragraph
+      // based on `interpreter-setting.json` config
+      Map<String, Object> config =
+          interpreterSettingManager.getConfigSetting(defaultInterpreterGroup);
+      paragraph.setConfig(config);
+    }
```

Comment:
Currently, the interpreter of new paragraph is not the default interpreter of note. It is the interpreter of the paragraph before the current paragraph. (we can discuss whether this is correct behavior). So in that case, I don't think you can set the default config of the default note interpreter.

Reply:
This is very important because:
When you create a new paragraph, you can directly display the custom configuration, for example, directly display the title bar, otherwise you must run this paragraph to display the title bar.
For example: create a new submarine interpreter note, the paragraph can directly display the title bar to facilitate the user to enter the algorithm name.

Reply:
I understand this. But as I said before, the default interpreter of new paragraph may not be the default interpreter of note. e.g. You can may get the config of `submarine` but the magics part could be `%python`. So I would suggest we first change the behavior of how to set the default interpreter of new paragraph. This is currently what zeppelin to do. https://github.com/apache/zeppelin/pull/3271/files#diff-0f40daff726ba29c9a004b342d422dfeR380

Reply:
There should be no problem. It has been shown in the GIF. You can modify the magic at any time. 
Assuming `submarine` sets the display title, `python` does not set the display title.
In the note created with submarine, all newly created paragraphs will display the title by default using the submarine configuration.
If you use `%python` to modify the default interpreter in a paragraph, Paragraph::applyConfigSetting() will re-use python's configuration to take effect in this paragraph.

The default interpreter behavior for each note, which was previously discussed in the email, may be a more complicated issue.